Toxygates/src/t/server/viewer/intermine/Intermine.scala

```diff
@@ -79,55 +79,35 @@ class IntermineConnector(instance: IntermineInstance,
     }
     println(s"${filtered take 100} ...")
 
-    Some(new StringList(StringList.PROBES_LIST_TYPE,
-        l.getName(), filtered.toArray))
+    Some(new StringList(StringList.PROBES_LIST_TYPE, l.getName(), filtered.toArray))
   }
 
   /**
    * Add a set of probe lists by first mapping them to genes
    */
-  def addProbeLists(ls: ListService,
-    lists: Iterable[StringList], replace: Boolean): Unit = {
-
-    for (l <- lists) {
-      addProbeList(ls, l.items(), Some(l.name()), replace)
+  def addProbeLists(ls: ListService, lists: Iterable[StringList], replace: Boolean): Unit = {
+    for { l <- lists } {
+      addProbeList(ls, l.items(), l.name(), replace)
     }
   }
 
   /**
-   * Obtain a valid list name, or None if the export cannot proceed.
-   * This potentially deletes a pre-existing list.
+   * Delete pre-existing lists with the same name as the one about to be exported, if replacement was requested.
+   * Returns a list name that may be used if the export can proceed, or None if it cannot.
    */
-  private def validNameForExport(ls: ListService, name: Option[String],
-      replace: Boolean): Option[String] = {
-
-    var serverList = name.map(n => Option(ls.getList(n))).flatten
-    if (serverList != None && replace) {
-      ls.deleteList(serverList.get)
-    }
+  private def checkListNameForExport(service: ListService, name: String, replace: Boolean): Option[String] = {
```

Comment:
I understand things are this way as an artifact of the old implementation, but my personal inclination would be to either split this into two methods, or to just move the logic into `addEntrezList`, because I feel it does two conceptually separate things:
- delete existing lists with the same name (if replace is true)
- return `name` as long as it's usable
  - (if this were going to be separated out into a function it would probably make more sense with a boolean return value)

Definitely not a high priority though, so I wouldn't object to this being merged as-is.

Reply:
Thanks for your comments, see what you think of the latest changes.